src/utils/gitlab-client.ts

```
@@ -0,0 +1,309 @@
+              break;
+            case 404:
+              reject(new GitLabApiError('Not Found: Resource does not exist', statusCode, errorResponse));
+              break;
+            case 422:
+              reject(new GitLabApiError(`Validation Error: ${errorMessage}`, statusCode, errorResponse));
+              break;
+            case 429:
+              const retryAfter = res.headers['retry-after'];
+              reject(
+                new GitLabApiError(`Rate Limit Exceeded. Retry after: ${retryAfter || '60'} seconds`, statusCode, {
+                  retryAfter: retryAfter ? parseInt(retryAfter as string, 10) : 60,
```

Biome (2.1.2)

[error] Other switch clauses can erroneously access this declaration.
Wrap the declaration in a block to restrict its access to the switch clause.

The declaration is defined in this switch clause:

Safe fix: Wrap the declaration in a block.


(lint/correctness/noSwitchDeclarations)